docs/reference/stores-guide/accumulo.md

```diff
@@ -0,0 +1,320 @@
+
+## Visibility
+
+Gaffer can take advantage of Accumulo's built-in fine-grained security to ensure that users only see data that they have authorisation to. This is done by specifying a "visibilityProperty" in the schema. This is a string that tells Accumulo which key in the `Properties` on each `Element` should be used for the visibility. The value of this property will be placed in the column visibility in each Accumulo key. This means that a user must have authorisations corresponding to that visibility in order to see the data.
+
+If no "visibilityProperty" is specified then the column visibility is empty which means that anyone who has read access to the table can view it.
+
+See [the visibility walkthrough](https://gchq.github.io/gaffer-doc/v1docs/getting-started/developer-guide/visibilities.html) in the [Dev Guide](https://gchq.github.io/gaffer-doc/v1docs/getting-started/developer-guide/contents.html) for an example of how properties can be aggregated over different visibilities at query time.
+
+## Timestamp
+
+Accumulo keys have a timestamp field. The user can specify which property is used for this by setting "timestampProperty" in the schema's config to the name of the property.
+If the timestamp is not set then it will be populated automatically:
+
+- Constant value - aggregated groups
+- Random number - non-aggregated groups
+
+Setting the timestamp yourself is an advanced feature and is not recommended as it can cause significant issues if it is not populated correctly.
+
+If you choose to set timestampProperty, the property will be aggregated with 'Max' - you cannot override this.
+
+## Validation and age-off of data
+
+In production systems where data is continually being ingested, it is necessary to periodically remove data so that the total size of the graph does not become too large. The `AccumuloStore` allows the creator of the graph to specify custom logic to decide what data should be removed. This logic is applied during compactions, so that the data is permanently deleted. It is also applied during queries so that even if a compaction has not happened recently, the data that should be removed is still hidden from the user.
+
+A common approach is simply to delete data that is older than a certain date. In this case, within the properties on each element there will be a time window specified. For example, the properties may contain a "day" property, and the store may be configured so that once the day is more than one year ago, it will be deleted. This can be implemented as follows:
+
+- Each element has a property called, for example, "day", which is a `Long` which contains the start of the time window. Every time an element is observed this property is set to the previous midnight expressed in milliseconds since the epoch.
+- In the schema the validation of this property is expressed as follows:
+```sh
```

Comment:
It is, but I think there's no JSON markup type and shell is the best fit. This was copied from the other repo, so it might be that there is a JSON highlighter now, I'll take a look.

Reply:
There's JSON support in MkDocs, so we don't need to use sh for this. I've fixed this now.